src/readiness.ts

```diff
@@ -0,0 +1,56 @@
+import { Bee, BeeDebug, Utils } from '@ethersphere/bee-js'
+import { READINESS_TIMEOUT } from './config'
+import { logger } from './logger'
+import { StampsManager } from './stamps'
+
+export async function checkReadiness(bee: Bee, beeDebug: BeeDebug, stampManager?: StampsManager): Promise<boolean> {
+  if (stampManager) {
+    const ready = await tryUploadingSingleChunk(bee, stampManager)
+
+    return ready
+  } else {
+    try {
+      const health = await beeDebug.getHealth({ timeout: READINESS_TIMEOUT })
+      const ready = health.status === 'ok'
+
+      return ready
+    } catch {
+      return false
+    }
+  }
+}
+
+async function tryUploadingSingleChunk(bee: Bee, stampsManager: StampsManager): Promise<boolean> {
+  const chunk = makeChunk()
+  try {
+    await bee.uploadChunk(stampsManager.postageStamp, chunk, { timeout: READINESS_TIMEOUT, deferred: true })
+
+    return true
+  } catch (error) {
```

Comment:
Are there some cases when this could fail and still be in a "valid state"?

Reply:
Not really, no. If we do not have postage stamp management, then it just tests the health endpoint. We can change that later to readiness endpoint on Bee as well.